py/server/deephaven/_udf.py

```diff
@@ -232,15 +234,18 @@ def _parse_signature(fn: Callable) -> _ParsedSignature:
         return _parse_np_ufunc_signature(fn)
     else:
         p_sig = _ParsedSignature(fn=fn)
-        sig = inspect.signature(fn, eval_str=True)
+        if sys.version_info.major == 3 and sys.version_info.minor >= 10:
+            sig = inspect.signature(fn, eval_str=True)
+        else:
+            sig = inspect.signature(fn)
```

Comment:
Does `__future__` provide a way to port this in a cleaner way?

Reply:
Not really. I think having an explicit version check actually helps future code cleanup when we bump the minimum supported Python version.